cpp/include/cudf/detail/transform.hpp

```diff
@@ -31,12 +31,14 @@ namespace detail {
  *
  * @param stream CUDA stream used for device memory operations and kernel launches.
  */
-std::unique_ptr<column> transform(column_view const& input,
-                                  std::string const& unary_udf,
-                                  data_type output_type,
-                                  bool is_ptx,
-                                  rmm::cuda_stream_view stream,
-                                  rmm::device_async_resource_ref mr);
+std::unique_ptr<column> transform(
+  std::vector<column_view> const& inputs,
+  std::string const& transform_udf,
```

Comment:
Maybe not something to worry about in this PR (though it would be nice) but is it possible to accept `cudf::host_span<column_view const> inputs` and `std::string_view const transform_udf` as it looks like we aren't really modifying them